### PR TITLE
[10.0][MIG] base_report_to_printer

### PR DESCRIPTION
**Pull request by Garamotte**
Migration of base_report_to_printer to v10.0
This migration includes improvements from #60

The deprecated methods in `printing.printer` have been removed (methode to update printers from cups, that is now done in `printing.server`).

### CODE REVIEW

**Comment by Garamotte** [comment]
@pedrobaeza @gurneyalex 
Do you know why Travis build crashes ? It seems that the PDF returned by wkhtmltopdf is invalid.
I tried other ways of installing wkhtmltopdf, but it didn't work.

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +20,11 @@ addons:
       - python-yaml
       - cups
       - libcups2-dev
+      - wkhtmltopdf  # only add if needed and check the before_install section below
+
+before_install:
+  - "export DISPLAY=:910.0"
```
Comment (pedrobaeza) [inline .travis.yml]: In the template we have `99'  instead of '910`

**Comment by Garamotte** [comment]
Ok, thanks, everything is OK now :)
I picked the value in another repo, and after a quick grep, I found 7 repos using 910 and only one using 99. Maybe wkhtmltopdf is not used in tests on these repos.

**Review by lasley — APPROVED** [review]
Code LGTM thanks @sylvain-garancher 

I'm installing this on a prod, so I'll report back if anything explodes. If I don't report back, we can assume this is also a functional 👍

**Comment by Garamotte** [comment]
I still have to split some tests, and edit some comments to follow guidelines, as you stated on my other PR a month ago, but this cannot break the module :)

**Comment by lasley** [comment]
Sweet thanks for the update @sylvain-garancher, and for making sure all your modules are top notch before merge 🚀 

So far our few tests on the prod box worked well, so functionally we're good in that respect too it seems.

**Comment by lasley** [comment]
@sylvain-garancher - FYI please make sure to include #73

**Comment by tedsalmon** [comment]
@lasley,

I have submitted a PR (https://github.com/syleam/report-print-send/pull/1) to forward-port these changes into @sylvain-garancher branch :)

**Comment by Garamotte** [comment]
Thanks, PR merged.

**Comment by Garamotte** [comment]
I'll include https://github.com/OCA/report-print-send/pull/74 when it's merged in 9.0.
Everything else OK on my side.

**Comment by lasley** [comment]
Long term test on my end is 👍  as well. #74 has been merged

**Comment by Garamotte** [comment]
I received a PR on the v10 migration branch : https://github.com/syleam/report-print-send/pull/2
Looking the issues list, I also found two issues and a PR for this bug on v9 : https://github.com/OCA/report-print-send/issues/75 / https://github.com/OCA/report-print-send/issues/63 / https://github.com/OCA/report-print-send/pull/67

I think the best would be to fix the bug in v9 (with a unit test, if possible, but I don't have time actually), then I'll cherry-pick the fix into my v10 migration branch.

**Comment by Garamotte** [comment]
@lasley Bug fix merged into 9.0 and cherry-picked here.
You can replace the "WIP" tag by "Needs review".

**Comment by lasley** [comment]
@sylvain-garancher Done! I'll also toss this tab in my to-review pile & come back to it soon 😄

**Review by lasley — APPROVED** [review]

**Comment by lasley** [comment]
@tedsalmon please review

**Comment by peterfbrandl** [comment]
hope I am right here:

after installing the module it is impossible to print a report in "HTML" directly to the screen. The whole system seems to be in an endless loop. In the background you can see the result of the html-report but the loading symbol is still there. But this is only after installing the report-to-printer module

Thank you for your help

screen_report

**Comment by lasley** [comment]
@peterfbrandl - thanks for the report. I can confirm this issue exists on Runbot:
* Install `sale` app
* Set quotation report type to HTML
* Navigate to a quote and print:
* HTML report is generated and shown, but the UI is never released. No error in console:
  * 
image


The error clears itself after uninstalling the module (the infinite loop will need to time out first, as well).

**Comment by Garamotte** [comment]
Thanks for giving a simple way to reproduce the issue.
I won't have time to look at this issue in the next days, any help is welcome :)

I think this is caused by the JS part : https://github.com/OCA/report-print-send/blob/4067f73e19e3238f13830b9e4d67b4af1d49f993/base_report_to_printer/static/src/js/qweb_action_manager.js
We see that `blockUI` is called in the beginning of the method, then the super is called because we don't use qweb-pdf here. Maybe the `unblockUI` method was called in the standard v9, and is no longer called on v10 ?

**Comment by Garamotte** [comment]
@peterfbrandl @lasley Fixed.
The UI is blocked (and then, unblocked) only when using qweb-pdf reports in v10, while it was blocked/unblocked for all reports on v9.
I simply moved the UI block call in the qweb-pdf case.

**Review by lasley — APPROVED** [review]
Confirmed fix, thanks @sylvain-garancher!

**Comment by peterfbrandl** [comment]
Confirmed fix. Thank you very much @sylvain-garancher

**Comment by lasley** [comment]
@peterfbrandl - Please use the Github review feature (in the Files Changed tab) so that we may count your review for merge.

**Comment by lasley** [comment]
I'm merging - we don't have the amount of reviews we need, but this has been in production for a while & I infer that @peterfbrandl is also using this as well - even though no explicit review was provided. 

Furthermore, we now have other contributors trying to upgrade this as well (#81). If something is broken & gets merged during this, I will take full responsibility and fix.

**Pull request merged**